libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt

```diff
@@ -77,27 +81,30 @@ class CarRouteLine(
         }
     }
 
+    private val navigationObserver = mapboxNavigationForward(this::onAttached, this::onDetached)
+
     private var styleLoadedListener: OnStyleLoadedListener? = null
 
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
         logAndroidAuto("CarRouteLine carMapSurface loaded $mapboxCarMapSurface")
         coroutineScope = MainScope()
-        val locationPlugin = mapboxCarMapSurface.mapSurface.location
+        carMapSurface = mapboxCarMapSurface
         styleLoadedListener = mapboxCarMapSurface.handleStyleOnAttached { style ->
-            val routeLineOptions = getMapboxRouteLineOptions(style)
+            val routeLineOptions = getMapboxRouteLineOptions(mapboxCarMapSurface.carContext, style)
             routeLineView = MapboxRouteLineView(routeLineOptions)
             routeLineApi = MapboxRouteLineApi(routeLineOptions).also {
                 routeLineView.initializeLayers(style)
             }
             routeArrowApi = MapboxRouteArrowApi()
             routeArrowView = MapboxRouteArrowView(
-                RouteArrowOptions.Builder(mainCarContext.carContext)
+                RouteArrowOptions.Builder(mapboxCarMapSurface.carContext)
                     .withAboveLayerId(TOP_LEVEL_ROUTE_LINE_LAYER_ID)
                     .build()
             )
 
+            val locationPlugin = mapboxCarMapSurface.mapSurface.location
             locationPlugin.addOnIndicatorPositionChangedListener(onPositionChangedListener)
-            mainCarContext.mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
+            MapboxNavigationApp.registerObserver(navigationObserver)
             coroutineScope.launch {
                 carRoutesProvider.navigationRoutes.collect { onRoutesChanged(it) }
             }
```

Comment:
Not related to this PR, but there seems to be an error, we launch a new job for every style load. We should also cancel the previous one.

Reply:
yeah this is a mess

Reply:
ok fixed it. nice catch!

Every time the style changed there is an extra call to `onRoutesChanged`, so if you change the style 6 times each re-route would trigger `onRoutesChanged` 6 times

Reply:
it's fixed